portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,17 +27,23 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  List<String> data = new ArrayList<>(); 
```

Comment:
Make this `private final` to avoid unnecessary accesses and mutations.

Note that even if the member itself is final, it is a collection so you can still add/remove entries.